nav2_mppi_controller/src/path_handler.cpp

```diff
@@ -44,11 +44,12 @@ PathHandler::getGlobalPlanConsideringBoundsInCostmapFrame(
   using nav2_util::geometry_utils::euclidean_distance;
 
   auto begin = global_plan_.poses.begin();
-  auto end = global_plan_.poses.end();
 
+  // Don't search further away than the prune distance
   auto closest_pose_upper_bound =
     nav2_util::geometry_utils::first_after_integrated_distance(
-    global_plan_.poses.begin(), global_plan_.poses.end(), max_robot_pose_search_dist_);
+    global_plan_.poses.begin(), global_plan_.poses.end(),
+        std::min(max_robot_pose_search_dist_, prune_distance_));
```

Comment:
I don't think prune distance has a role here in deciding where to look for the closest point on the path. It is not related to the pruning task, it is trying to find the closest point on the path if there is or is not regular replanning.

> >  Distance ahead of nearest point on path to robot to prune path to.

> I believe it is actually the max integrated distance/path distance to look for the closest point to the robot, from the first pose of the pruned global path

I agree this is a wrong description, where is that from? I write most of the docs quickly in one sitting so I probably copy pasted wrong.

Reply:
> I don't think prune distance has a role here in deciding where to look for the closest point on the path. It is not related to the pruning task, it is trying to find the closest point on the path if there is or is not regular replanning. If someone hasn't replanned recently, I'm not sure by the prune distance would impact the search distance.

We need to limit our search for the closest point up to the prune_distance. Otherwise, because the closest point is used as the lower bound in the search for the upper bound of the plan to transform, we could have the closest point further along the path that the `prunned_plan_end` and hence the lower bound higher than the upper bound.

```
  // Find the furthest relevent pose on the path to consider within costmap
  // bounds
  // Transforming it to the costmap frame in the same loop
  for (auto global_plan_pose = closest_point; global_plan_pose != pruned_plan_end;
    ++global_plan_pose)
  {
```


> I agree this is a wrong description, where is that from? I write most of the docs quickly in one sitting so I probably copy pasted wrong.
Description has been updated compared to what I was looking at, but still doesn't reflect that it is the distance from the first point of the global plan (which is pruned at each iteration)

https://github.com/ros-planning/navigation2/blob/29ad43b96097383e90b7b5c24594476b6990b09b/nav2_mppi_controller/README.md?plain=1#L64

Reply:
> We need to limit our search for the closest point up to the prune_distance. Otherwise, because the closest point is used as the lower bound in the search for the upper bound of the plan to transform, we could have the closest point further along the path that the prunned_plan_end and hence the lower bound higher than the upper bound.

I don't think that's accurate? If we have a path that we're just starting (so path marker 0 is the closest point) then we are searching forward from there by the `prune_distance` to find the last point of interest for us forward.  If we instead had path marker 22 as the closest point, the same logic applies. No matter what path marker is chosen as the closest point, we're integrating **from that pose** by the `prune_distance`. If we selected a path pose further from marker 0 than the `prune_distance`, that's unimportant (since we then prune forward **from** that pose by the distance). If you wanted to limit how far we look ahead from the last point, that's what `max_robot_pose_search_dist` is for.

From this line

```
  auto pruned_plan_end =
    nav2_util::geometry_utils::first_after_integrated_distance(
    closest_point, global_plan_.poses.end(), prune_distance_);
 ```

I don't see how its possible that `closest_point` could ever be further along the path than `pruned_plan_end`. By definition here, its at least `closest_point` or further.

Reply:
You are totally right, I got confused